### ISSUE TITLE
iOS stops building when using Segment's @import with `-fcxx-modules` C++ flag

### ISSUE THREAD

**Issue by jkadamczyk** [segmentio/analytics-react-native#609]
I was trying to upgrade `analytics-react-native` library from 1.5.0 to 2.3.2. Everything works apart from iOS builds when you add `@import` in the `AppDelegate.mm` file. When you add it XCode will start telling you that you need to enable Objective-C++ modules. After doing so in a fresh 0.68.1 app it works fine. Problem arises when you have any custom ViewManagers (Custom Native Views) implemented like we have. If that's the case, somehow the Yoga library stops building saying `BitUtils.h` file is not found.

I was thinking it could be possible to mitigate effect of this if the segment library allowed using `#import` instead of `@import` since that would eradicate the need of using `-fcxx-modules` flag and probably make it work on all newer React Native versions where people use custom native views.

- `analytics-react-native` version: 2.3.2
- Integrations versions (if used): none
- React Native version: 0.68.1
- iOS or Android or both? iOS

**Steps to reproduce**

- Create a new fresh React Native project with version `0.68.1`
- Install segment libraries like mentioned in docs
- Add code to AppDelegate.mm file to `trackDeepLinks`
- Build -> should build just fine
- Create a new header file named for example Testing.h (simulates ProjectName-Swift.h file that includes carious other headers used like declarations of Views written in Swift) with #import <React/RCTConvert.h> in it
- Import the new Testing.h header into AppDelegate.mm file
- The app should stop building saying that it cannot build Yoga.

**Expected behavior**

- The application should be building fine with recent React Native versions

**Actual behavior**

- The application won't build in a project with recent React Native version that is more complicated than a simple example app

Let me know if I can help in some way to visualize the issue.

tl;dr I would love to import the library in a different way than `@import segment_analytics_react_native` like `#import ...`

**Comment by osdnk** [comment]
Same here

**Comment by jkadamczyk** [comment]
We managed to fix it with some precompiler macros to ignore some files in C++ but would be cool if there was an option to not use Objective C modules at all and if it was mentioned as an option in the docs. For new projects adding the import from your docs requires C++ compiler flags modifications which a lot of devs working with React Native won't be comfortable about.

**Comment by jkadamczyk** [comment]
For anyone else interested
https://github.com/rainbow-me/rainbow/pull/3748

**Comment by alanjcharles** [comment]
Thanks @jkadamczyk! fixed in `2.5.0`. Please let us know if you run into any other issues.

**Comment by jkadamczyk** [comment]
Will try and report back! Thanks for working on that!

**Comment by alanjcharles** [comment]
np! Facebook themselves also pushed a similar update on Monday if you're interested: https://github.com/facebook/react-native/releases/tag/v0.69.3
